styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java

```diff
@@ -171,7 +193,9 @@ private void ensureSecret(WorkflowInstance workflowInstance, RunSpec runSpec) {
                                     .build());
         LOG.info("[AUDIT] Secret {} created for service account {}", secretName, serviceAccount);
       }
-    } else if (runSpec.secret().isPresent()) {
+    }
+
+    if (runSpec.secret().isPresent()) {
```

Comment:
Old-style secrets still make sense for OSS I think. Smaller organizations might want to use k8s secrets.

For us I think having both allows users to migrate their workflows. Not allowing both makes it harder because you have to make sure to change both the container and the workflow config at the same time.